### PR TITLE
[RUM 6317] Add documentation for view context feature

### PR DESCRIPTION
**Pull request by cy-moi**
### What does this PR do? What is the motivation?

Add documentation for custom view specific context with current and new APIs.
### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/real_user_monitoring/browser/advanced_configuration.md [hunk 1]
```diff
@@ -92,35 +93,82 @@ Starting with [version 2.17.0][3], you can add view names and assign them to a d
...
-         name: 'checkout',
-         service: 'purchase',
-         version: '1.2.3'
-       })
+      window.DD_RUM.startView({
+        name: 'checkout',
+        service: 'purchase',
+        version: '1.2.3',
+        context: {
+          team: 'payment'
```
Comment (amortemousque) [inline content/en/real_user_monitoring/browser/advanced_configuration.md:114]: **💬 suggestion:** `‏team` is specific a concept in Datadog. I think we should another example: Maybe
```
context: {
          payment: 'Done'
```

**Review by thomas-lebeau — APPROVED** [review]

File: content/en/real_user_monitoring/browser/advanced_configuration.md (hunk 1 above)
Comment (cy-moi) [reply content/en/real_user_monitoring/browser/advanced_configuration.md]: Changed it to this ⬆️

File: content/en/real_user_monitoring/browser/advanced_configuration.md [hunk 2]
```diff
@@ -793,6 +841,108 @@ acceptCookieBannerButton.addEventListener('click', () => {
 {{% /tab %}}
 {{< /tabs >}}
 
+## View context
+
+Starting with [version 5.28.0][21], the contexts of view events are modifiable. The context adds to the current view only and populate its children events (such as `action`, `error` and `timing`) with `startView`, `setViewContext` and `setViewContextProperty`.
```
Comment (buraizu) [inline content/en/real_user_monitoring/browser/advanced_configuration.md:846]:
```suggestion
Starting with [version 5.28.0][21], the context of view events is modifiable. Context can be added to the current view only, and populates its child events (such as `action`, `error`, and `timing`) with `startView`, `setViewContext`, and `setViewContextProperty` functions.
```

Comment (buraizu) [reply content/en/real_user_monitoring/browser/advanced_configuration.md:846]: Attempted to clarify this sentence, let me know if any of the suggestions above are inaccurate.

File: content/en/real_user_monitoring/browser/advanced_configuration.md [hunk 3]
```diff
@@ -793,6 +841,108 @@ acceptCookieBannerButton.addEventListener('click', () => {
...
+
+### Start view with context
+
+Optionally define the context while starting view with [`startView` options](#override-default-rum-view-names).
```
Comment (buraizu) [inline content/en/real_user_monitoring/browser/advanced_configuration.md:850]:
```suggestion
Optionally define the context while starting a view with [`startView` options](#override-default-rum-view-names).
```

File: content/en/real_user_monitoring/browser/advanced_configuration.md [hunk 4]
```diff
@@ -793,6 +841,108 @@ acceptCookieBannerButton.addEventListener('click', () => {
...
+
+### Add view context
+
+Enrich or modify the context of RUM view events and corresponding children events with  `setViewContextProperty(key: string, value: any)` API.
```
Comment (buraizu) [inline content/en/real_user_monitoring/browser/advanced_configuration.md:854]:
```suggestion
Enrich or modify the context of RUM view events and corresponding child events with the `setViewContextProperty(key: string, value: any)` API.
```

File: content/en/real_user_monitoring/browser/advanced_configuration.md [hunk 5]
```diff
@@ -459,7 +507,7 @@ You can update the following event properties:
 |   `error.stack `        |   String  |   The stack trace or complementary information about the error.                                     |
 |   `error.resource.url`  |   String  |   The resource URL that triggered the error.                                                        |
 |   `resource.url`        |   String  |   The resource URL.                                                                                 |
-|   `context`        |   Object  |   Attributes added with the [Global Context API](#global-context) or when generating events manually (for example, `addError` and `addAction`). RUM view events `context` is read-only.                                                                                 |
+|   `context`        |   Object  |   Attributes added with the [Global Context API](#global-context), then [View Context API](#view-context) or when generating events manually (for example, `addError` and **`addAction`**).                                                                                 |
```
Comment (buraizu) [inline content/en/real_user_monitoring/browser/advanced_configuration.md:510]:
```suggestion
|   `context`        |   Object  |   Attributes added with the [Global Context API](#global-context), the [View Context API](#view-context), or when generating events manually (for example, `addError` and **`addAction`**).                                                                                 |
```

File: content/en/real_user_monitoring/browser/advanced_configuration.md [hunk 6]
```diff
@@ -793,6 +841,108 @@ acceptCookieBannerButton.addEventListener('click', () => {
...
+    amount: 23.42
+});
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+
+### Replace view context
+
+Replace the context of your RUM view events and corresponding children events with `setViewContext(context: Context)` API.
```
Comment (buraizu) [inline content/en/real_user_monitoring/browser/advanced_configuration.md:901]:
```suggestion
Replace the context of your RUM view events and corresponding child events with `setViewContext(context: Context)` API.
```

File: content/en/real_user_monitoring/browser/advanced_configuration.md [hunk 7]
```diff
@@ -81,6 +81,7 @@ Starting with [version 2.17.0][3], you can add view names and assign them to a d
    - View Name: Defaults to the page URL path.
    - Service: Defaults to the default service specified when creating your RUM application.
    - Version: Defaults to the default version specified when creating your RUM application.
+   - Context: Starting with [version 5.28.0][21], you can add context to view and view children events.
```
Comment (buraizu) [inline content/en/real_user_monitoring/browser/advanced_configuration.md:84]:
```suggestion
   - Context: Starting with [version 5.28.0][21], you can add context to views and the child events of views.
```

File: content/en/real_user_monitoring/browser/advanced_configuration.md [hunk 8]
```diff
@@ -1035,3 +1185,6 @@ However, this feature comes with some **limitations**:
 [17]: https://github.com/DataDog/browser-sdk/blob/main/CHANGELOG.md#v4130
 [18]: /data_security/real_user_monitoring/#browser-rum-use-of-cookies
 [19]: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage
+[20]: /real_user_monitoring/browser/advanced_configuration/?tab=npm#override-default-rum-view-names
```
Comment (buraizu) [inline content/en/real_user_monitoring/browser/advanced_configuration.md:1188]: It looks like this link (`[20]`) didn't get added anywhere. Was it meant to be added somewhere?

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for the PR, just requesting some minor updates and clarifications.

File: content/en/real_user_monitoring/browser/advanced_configuration.md (hunk 8 above)
Comment (cy-moi) [reply content/en/real_user_monitoring/browser/advanced_configuration.md]: I removed this link and updated the other one.

**Review by buraizu — APPROVED** [review]
Thanks, LGTM. It looks like you're still waiting on one more reviewer, so let me know when it's ready to merge.

**Comment by cy-moi** [comment]
> Thanks, LGTM. It looks like you're still waiting on one more reviewer, so let me know when it's ready to merge.

Hi thank you. It's ready to merge.

**Pull request merged**